execution/gethexec/forwarder.go

```diff
@@ -157,28 +173,44 @@ func (f *TxForwarder) CheckHealth(inctx context.Context) error {
 		}
 		ctx, cancelFunc := context.WithTimeout(context.Background(), timeout)
 		defer cancelFunc()
-		f.healthErr = f.rpcClient.CallContext(ctx, nil, "arb_checkPublisherHealth")
+		f.healthErr = f.rpcClients[0].CallContext(ctx, nil, "arb_checkPublisherHealth")
 		f.healthChecked = time.Now()
 	}
 	return f.healthErr
 }
 
 func (f *TxForwarder) Initialize(inctx context.Context) error {
-	if f.target == "" {
-		f.rpcClient = nil
-		f.ethClient = nil
-		f.enabled.Store(false)
-		return nil
+	if f.ctx == nil {
+		f.ctx = inctx
+	}
+
+	pos := 0
+	for pos < len(f.targets) {
+		if f.targets[pos] == "" {
+			f.targets = append(f.targets[:pos], f.targets[pos+1:]...)
+		} else {
+			pos += 1
+		}
 	}
-	ctx, cancelFunc := f.ctxWithTimeout(inctx)
+
+	ctx, cancelFunc := f.ctxWithTimeout()
 	defer cancelFunc()
-	rpcClient, err := rpc.DialTransport(ctx, f.target, f.transport)
-	if err != nil {
-		return err
+	pos = 0
+	for pos < len(f.targets) {
+		rpcClient, err := rpc.DialTransport(ctx, f.targets[pos], f.transport)
+		if err != nil {
+			log.Warn("error initializing a forwarding client in txForwarder", "forwarding url", f.targets[pos], "err", err)
+			f.targets = append(f.targets[:pos], f.targets[pos+1:]...)
+			continue
+		}
+		ethClient := ethclient.NewClient(rpcClient)
+		f.rpcClients = append(f.rpcClients, rpcClient)
+		f.ethClients = append(f.ethClients, ethClient)
+		pos += 1
+	}
```

Comment:
[nit] lets use something like this 

```
	ctx, cancelFunc := f.ctxWithTimeout()
	defer cancelFunc()
	var targets []string
	for _, target := range f.targets {
		if target == "" {
			continue
		}
		rpcClient, err := rpc.DialTransport(ctx, target, f.transport)
		if err != nil {
			log.Warn("error initializing a forwarding client in txForwarder", "forwarding url", f.targets[pos], "err", err)
			continue
		}
		targets = append(targets, target)
		ethClient := ethclient.NewClient(rpcClient)
		f.rpcClients = append(f.rpcClients, rpcClient)
		f.ethClients = append(f.ethClients, ethClient)
	}
	f.targets = targets
```

existing logic seems to be a little hard to follow, though let me know if the existing logic is more efficient in some manner

Reply:
Agreed!